src/components/Menu/menu.js

```diff
@@ -1,14 +1,38 @@
 import React from 'react';
 import './menu.css';
+import { useDispatch } from 'react-redux';
+import { fetchMovies } from '../../reducers/movieSlice';
 
-const Menu = () => (
-  <ul className="menu">
-    <li><a href="/">All</a></li>
-    <li><a href="/">Documentary</a></li>
-    <li><a href="/">Comedy</a></li>
-    <li><a href="/">Horror</a></li>
-    <li><a href="/">Crime</a></li>
-  </ul>
-);
+const Menu = () => {
+    const dispatch = useDispatch();
+    
+    function fetchWithFilter(filterValue) {
+      return () => dispatch(fetchMovies({
+        params: {
+          filter: filterValue,
+        },
+      }));
+    }
+    
+    return (
+      <ul className="menu">
+        <li>
+            <button onClick={() => dispatch(fetchMovies())}>All</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Documentary')}>Documentary</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Comedy')}>Comedy</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Horror')}>Horror</button>
+        </li>
+        <li>
+            <button onClick={fetchWithFilter('Crime')}>Crime</button>
+        </li>
```

Comment:
I bet that one can be rendered in a loop

Reply:
fixed